app/controller/grid/Grid.js

```diff
@@ -688,5 +703,151 @@ Ext.define('CpsiMapview.controller.grid.Grid', {
         };
 
         viewModel.setStores(stores);
+    },
+
+    /**
+     * Activated the "preset filter" button if the layer
+     * has the repspective properties.
+     *
+     * @param {Ext.app.ViewModel } viewModel The ViewModel
+     */
+    activatePresetFilterButton: function (viewModel) {
+        var me = this;
+        // check if layer has preset grid Filters
+        // if yes, we activate the respective button
+        var wmsLayerKey = viewModel.get('wmsLayerKey');
+        var vectorLayerKey = viewModel.get('vectorLayerKey');
+        var layer;
+        if (wmsLayerKey) {
+            layer = me.getLayerByKey(wmsLayerKey);
+        } else if (vectorLayerKey) {
+            layer = me.getLayerByKey(vectorLayerKey);
+        }
+        if (layer && layer.get('gridFilters')) {
+            viewModel.set('usePresetFilters', true);
+        }
+    },
+
+    /**
+     * Applies preset filters from the configuration
+     * to the grid.
+     */
+    onApplyPresetFilters: function () {
+        var me = this;
+
+        var viewModel = me.getViewModel();
+
+        var wmsLayerKey = viewModel.get('wmsLayerKey');
+        var vectorLayerKey = viewModel.get('vectorLayerKey');
+
+        var layer;
+        if (wmsLayerKey) {
+            layer = me.getLayerByKey(wmsLayerKey);
+        } else if (vectorLayerKey) {
+            layer = me.getLayerByKey(vectorLayerKey);
+        } else {
+            // no layer found
+            return;
+        }
+        var gridFilters = layer.get('gridFilters');
+
+        if (!gridFilters || !Ext.isArray(gridFilters)) {
+            return;
+        }
+
+        var grid = me.getView();
+        if (!grid) {
+            return;
+        }
+        var columnManager = grid.getColumnManager();
+        if (!columnManager) {
+            return;
+        }
+
+        // loop through all provided preset filter definitions
+        Ext.each(gridFilters, function (filterDef) {
+            if (!filterDef || !Ext.isObject(filterDef)) {
+                return;
+            }
+
+            var columnName = filterDef.property;
+            var value = filterDef.value;
+            var operator = filterDef.operator;
+
+            if (!columnName || !value || !operator) {
+                Ext.log.warn('Preset filter is not properly defined.');
+                Ext.log.warn(filterDef);
+                return;
+            }
+
+            var column = columnManager.getHeaderByDataIndex(columnName);
+            if (!column || !column.filter || !column.filter.type) {
+                return;
+            }
+            var columnType = column.filter.type;
+
+            switch (columnType) {
+                case 'string':
+                    // only equal is supported for string
+                    if (operator !== '=') {
+                        return;
```

Comment:
Would it make sense to print a warning here?

Reply:
Good point. I added a warning with 77c7639